### PR TITLE
test: extend unit test for dynamo-route-caching-provider

### PR DESCRIPTION
**Pull request by lynnshaoyu**
Adds unit tests for dynamo-route-caching-provider file for all 3 modes of dynamo route cache: LiveMode (cache is on), TapCompare (cache is on as well, but use original routing, to compare the efficacy of the two), and DarkMode( cache is off).

### CODE REVIEW

**Comment by jsy1218** [comment]
Seeing:

> This branch is out-of-date with the base branch

Might need a rebase.

**Comment by jsy1218** [comment]
consideration: I noticed that the routine-api can be noticeably slow.. To establish first line of perf defense, we may consider test timeouts as caching layer SLA (assuming there's no test timeout here). This is once we think the caching layer perf is stably good.

File: test/unit/handlers/router-entities/route-caching/dynamo-route-caching-provider.test.ts [hunk 1]
```diff
@@ -4,52 +4,218 @@ import 'reflect-metadata'
...
-  'Uniswap'
+const WETH = WNATIVE_ON(ChainId.MAINNET)
+
+const TEST_WETH_USDC_POOL = new Pool(
+  WETH,
+  USDC_MAINNET,
+  FeeAmount.HIGH,
+  '2437312313659959819381354528',
+  '10272714736694327408',
+  -69633
```
[the remark this answers is not in the document]
Comment (mikeki) [reply test/unit/handlers/router-entities/route-caching/dynamo-route-caching-provider.test.ts:59]: I agree, but also I don't think these  would really impact the code, they are just required fields in the Pool class

**Review by mikeki — APPROVED** [review]

File: test/unit/handlers/router-entities/route-caching/dynamo-route-caching-provider.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (lynnshaoyu) [reply test/unit/handlers/router-entities/route-caching/dynamo-route-caching-provider.test.ts:59]: done, added inline comments for the param names

**Pull request merged**